exporter/signalfxexporter/exporter.go

```diff
@@ -106,36 +108,55 @@ func newSignalFxExporter(
 		return nil, fmt.Errorf("failed to create metric converter: %w", err)
 	}
 
-	transport := http.DefaultTransport.(*http.Transport).Clone()
-	transport.MaxIdleConns = config.MaxConnections
-	transport.MaxIdleConnsPerHost = config.MaxConnections
-	transport.IdleConnTimeout = 30 * time.Second
+	return &signalfxExporter{
+		config:            config,
+		logger:            createSettings.Logger,
+		telemetrySettings: createSettings.TelemetrySettings,
+		converter:         converter,
+	}, nil
+}
+
+func (se *signalfxExporter) start(_ context.Context, host component.Host) (err error) {
+	options, err := se.config.getOptionsFromConfig()
+	if err != nil {
+		return err
+	}
+
+	headers := buildHeaders(se.config)
+	se.config.Headers = nil
```

Comment:
Line 127 - Line 137  https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16837/files#diff-8263390249c86b6a1730e8dc0732629a1c7e707797f96ad5745a24e11d284f85R127-R137
seems same as 
Line 222 - Line 234 https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16837/files#diff-8263390249c86b6a1730e8dc0732629a1c7e707797f96ad5745a24e11d284f85R222-R234

Maybe convert those logic into a common function is better?

Reply:
Good idea, done!